discovery/discovery/src/main/java/io/helidon/discovery/Discovery.java

```diff
@@ -0,0 +1,205 @@
+ * for the invocation. If the discovered set would otherwise be {@linkplain SequencedSet#isEmpty() empty}, it will
+ * instead contain at least a {@link DiscoveredUri} whose {@link DiscoveredUri#uri() uri()} method returns the default
+ * value supplied to the invocation of the {@link #uris(String, URI)} method that produced it. More characteristics and
+ * requirements of discovery names, discovered sets, and discovered URIs may be found below.</p>
+ *
+ * <h2>Acquisition</h2>
+ *
+ * <p>To acquire a {@link Discovery} instance, <a
+ * href="https://helidon.io/docs/latest/se/injection#_injection_points">inject</a> an instance of {@link Discovery},
+ * using the {@link io.helidon.service.registry.Service.Inject} annotation. Alternatively, and equivalently, use the
+ * {@code static} {@link io.helidon.service.registry.Services#get(Class)} method, supplying {@code Discovery.class} as
+ * its sole argument.</p>
+ *
+ * <h2>Discovery Names</h2>
+ *
+ * <p>A discovery name is a name for which a {@link Discovery} implementation is capable of furnishing, at a moment in
+ * time, zero or more URIs, represented by {@link DiscoveredUri} instances.</p>
+ *
+ * <p>A {@link Discovery} implementation <em>must</em> be prepared to handle names that do not identify any URIs at the
+ * moment of invocation, or will never identify any URIs. In either case, a {@link Discovery} implementation
+ * <em>must</em> return a discovered set that consists solely of a {@link DiscoveredUri} that represents the default
+ * value supplied to the invocation of the {@link #uris(String, URI)} method that produced it. Implementations
+ * <em>should</em> return a {@link DiscoveredUri} in the {@linkplain SequencedSet#getLast() tail position} of the
+ * discovered set in all cases.</p>
+ *
+ * <p>For maximum portability, a discovery name <em>should</em> comprise only <a
+ * href="https://www.rfc-editor.org/rfc/rfc1123#page-13">legal hostname characters</a> (to permit DNS-based or DNS-using
+ * implementations).</p>
+ *
+ * <p>Discovery implementations <em>should</em> treat discovery names case-insensitively.</p>
+ *
+ * <p>Since a discovery name is an application-level concern, not a component-level comncern, it is recommended that
+ * <dfn>discovery-using components</dfn>&mdash;which may be included in disparate applications with disparate naming
+ * requirements and {@link Discovery} implementations&mdash;allow an application developer to configure, or otherwise
+ * specify at deployment or runtime, any relevant discovery names that they use to account for potential differences in
+ * application and component namespaces.</p>
+ *
+ * <h2>Discovered Sets</h2>
+ *
+ * <p>Invocations of the {@link #uris(String, URI)} method return immutable {@link SequencedSet}s of {@link
+ * DiscoveredUri}s. These are <dfn>discovered sets</dfn>. Discovered sets have at least the following
+ * characteristics:</p>
+ *
+ * <ol>
+ *
+ * <li>They are <dfn>immutable</dfn>, following the <a
```

Comment:
Hmm. If returned discovered sets are _unmodifiable_ then the _caller_ cannot change them but it's conceivable that the backing data structure _could_ change so they are not necessarily _immutable_. Maybe clarify what the contract requires or expects of implementations on this score?

Reply:
Not trying to shut you down. If you have better wording in mind I'm willing to apply it!

Reply:
No; what's there conveys what's needed, I think. I was just desperate to find _something_ to comment about!